### PR TITLE
chore: [CDS-78488]: add matrixMetadata information in the stage/step popover in the console and visual view

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

**Issue**: Matrix metadata information was being shown in the node names on the visual and console view causing discrepancy as the axis information can be very huge in some cases of Policy Step, multi-svc, multi-infra cases.

**Solution**: Name/identifier is the single source of truth for the names of the node coming from the BE ( for matrix identifiers have labels appended if the ' Enable matrix labels by the name` setting is enabled ). 
Additionally,
 - Metadata information for the stage/step will reside in the popover containing additional information
 - Console view supports dynamic popover for stage/step name similar to the visual view
 
 **Scenarios tested:**
 
 
 ## Stage nodes
 
 **1. Matrix nodes**
 
 | Visual | Console |
 | -- | -- |
 | image | image |
| image| image |

**2. Multi service**

 | Visual | Console |
 | -- | -- |
| image | image |

**3. Multi service/multi-infra**

 | Visual | Console |
 | -- | -- |
| image | image |

**4. Multi-service/multi-env**
 
  | Visual | Console |
 | -- | -- |
| image | image |

**6. Repeat strategy**

 | Visual | Console |
 | -- | -- |
 | image | image |

**7. Resource Constraint Conditional Execution**

 | Visual | Console |
 | -- | -- |
 | image | image | 

 ## Step nodes

**1. Matrix node**
 | Visual | Console |
 | -- | -- |
 | image | image |

**2. Matrix stepGroup node**
 | Visual | Console |
 | -- | -- |
 | image | image |

**3. Repeat node**
 | Visual | Console |
 | -- | -- |
 | image | image |

**4. Hostname node**
 | Visual | Console |
 | -- | -- |
|image | image |

Video:

https://github.com/harness/harness-core-ui/assets/100121280/03cf08ab-35be-4430-8a9a-5b9cc0736167

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionLogView/StageSelection/__tests__/StageSelection.test.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  useExecutionContext: () => executionContextMock
+}))
+
+const assertEntityPopover = async (container: HTMLElement, entityName: string): Promise<void> => {
+  await act(async () => {
+    fireEvent.mouseEnter(getAllByText(container, entityName)[0])
+  })
+  const dynamicPopover = document.querySelector('[class*="dynamicPopover"]')
+  expect(dynamicPopover).toBeDefined()
+  await waitFor(() => expect(document.body.querySelector('[data-testid="dynamic-popover"]')).toBeInTheDocument())
```
[the remark this answers is not in the document]
Comment (PratyushGar) [reply src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionLogView/StageSelection/__tests__/StageSelection.test.tsx]: Done

**Comment by krishi0408** [comment]
trigger messagecheck

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionLogView/StageSelection/__tests__/StageSelection.test.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+jest.mock('@pipeline/context/ExecutionContext', () => ({
+  useExecutionContext: () => executionContextMock
+}))
+
+const assertEntityPopover = async (container: HTMLElement, entityName: string): Promise<void> => {
+  await act(async () => {
+    fireEvent.mouseEnter(getAllByText(container, entityName)[0])
+  })
+  const dynamicPopover = document.querySelector('[class*="dynamicPopover"]')
+  expect(dynamicPopover).toBeDefined()
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionLogView/StageSelection/__tests__/StageSelection.test.tsx:23]: Could this be removed, as line 24 has a better assertion using test ID?

File: src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionLogView/StageSelection/StageSelection.module.scss [hunk 3]
```diff
@@ -113,30 +113,10 @@
   cursor: pointer;
 }
 
-.name {
-  white-space: nowrap !important;
+.entityName {
+  display: -webkit-box !important;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: 1;
```
[the remark this answers is not in the document]
Comment (PratyushGar) [reply src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionLogView/StageSelection/StageSelection.module.scss:119]: Yes, in stage selection we are suppressing the line clamp tooltip and showing a dynamic popover with name and metadata inside. So removed the inline line clamp property and used CSS for the same, as uicore always shows a tooltip if line-clamp is associated.

**Review by satyam-harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**